assets/app/scripts/controllers/edit/buildConfig.js

```diff
@@ -633,21 +633,30 @@ angular.module('openshiftConsole')
     };
 
     $scope.updateTriggers = function() {
+
+      // helper for creating a webhook object
+      function createWebhook(type) {
+        var webhook = {
+          type: type
+        };
+        webhook[(type === "GitHub") ? "github" : "generic"] = {
+          secret: ApplicationGenerator._generateSecret()
+        };
+        return webhook;
+      }
       var presentTriggers = $scope.triggers.present;
       var triggers = [];
+
       if (presentTriggers.webhook) {
-        var webhooks = _.filter($scope.buildConfig.spec.triggers, function(obj) { return obj.type === "GitHub" })
+        var webhooks = _.filter($scope.buildConfig.spec.triggers, function(obj) { return obj.type === "GitHub" || obj.type === "Generic"})
         if (webhooks.length === 0) {
-          webhooks.push({
-            github: {
-              secret: ApplicationGenerator._generateSecret()
-            },
-            type: "GitHub"
-          });
+          webhooks.push(createWebhook("GitHub"));
+          webhooks.push(createWebhook("Generic"));
+        } else if (webhooks.length === 1) {
```

Comment:
hmm, this makes me think the editor should really have these as separate options.  if i choose to go in and delete one of them in the yaml editor, then i dont want it to add it back any time I use the GUI editor.